src/main/java/com/aws/greengrass/componentmanager/ComponentStore.java

```diff
@@ -192,7 +193,8 @@ Optional<String> findComponentRecipeContent(@NonNull ComponentIdentifier compone
         }
 
         try {
-            return Optional.of(new String(Files.readAllBytes(recipePath), StandardCharsets.UTF_8));
+            String recipeContent = new String(Files.readAllBytes(recipePath), StandardCharsets.UTF_8);
+            return Utils.isEmpty(recipeContent) ? Optional.empty() : Optional.of(recipeContent);
```

Comment:
This will result in the plugin being treated as unloadable but the error message will say that the plugin has been modified - which implies modified by someone intentionally, rather than saying the recipe is corrupt/invalid. Should we have an exception thrown from the validateDigest method that can be used to communicate the problem better?

Reply:
updated the pr to clarify the error message